hedera-node/src/main/java/com/hedera/services/store/StoresModule.java

```diff
@@ -88,7 +100,8 @@
 				NftProperty.class,
 				MerkleUniqueToken::new,
 				backingNfts,
-				new ChangeSummaryManager<>());
+				new ChangeSummaryManager<>(),
+				new AccountsCommitInterceptor(new SideEffectsTracker()));
```

Comment:
I think we'll need to work with a singleton instance of the `AccountsCommitInterceptor` which is bound to the global singleton instance of the `SideEffectsTracker` which is also used by `BasicTransactionalContext` for top level record creation. So here we don't want to create new instances of `AccountsCommitInterceptor` and `SideEffectsTracker`.
But we'll also need to make any given `Updater`'s `WorldLedger` instance's `TransactionalLedgers`'s `AccountsCommitInterceptors` all work with the `SideEffectsTracker` instance created by the `HTSPrecompiledContract` `sideEffectsFactory`. We need to figure out the best way to ensure this.

cc @tinker-michaelj can you confirm my thinking is correct?

Reply:
I missed this comment, but exactly @stoqnkpL! 👍